src/main/scala/com/twitter/finatra/http/SwaggerRouteDSL.scala

```diff
@@ -11,57 +11,68 @@ object SwaggerRouteDSL {
   implicit def convertToSwaggerRouteDSL(dsl: RouteDSL)(implicit swagger: Swagger): SwaggerRouteDSL = new SwaggerRouteDSLWapper(dsl)(swagger)
 }
 
+object PathHelper {
+  implicit class PathWrapper(path: String) {
+    def tailed() = s"/$path/".replaceAll("/{2,}", "/")
+  }
+  implicit class SwaggerWrapper(swagger: Swagger) {
+    def base() = Option(swagger.getBasePath).getOrElse("/")
+  }
+}
+
 trait SwaggerRouteDSL {
+  import PathHelper._
+
   implicit protected val swagger: Swagger
   protected val dsl: RouteDSL
 
   def postWithDoc[RequestType: Manifest, ResponseType: Manifest](route: String, name: String = "", admin: Boolean = false, adminIndexInfo: Option[AdminIndexInfo] = None)
                                                                 (doc: Operation => Unit)
                                                                 (callback: RequestType => ResponseType): Unit = {
-    registerOperation(route, "post")(doc)
-    dsl.post(route, name, admin, adminIndexInfo)(callback)
+    registerOperation(route.tailed, "post")(doc)
```

Comment:
The changes are mainly to apply the [`basePath`](http://swagger.io/specification/#swaggerBasePath) setting to the registered Finatra routes.
 
And the `tailed` function is, for my personal convenience, to ensure that there is no consecutive occurrences of `/` in the registered routes, and every route will be tailed with a `/`.  As an example, if the base path is given as `/v1/` and the route is given as `/student`, the `tailed` route will be `/v1/student/` rather than the original `/v1//student`.

To be honest, the `tailed` function is not really necessary and I will remove it.